cases/widgets.py

```
@@ -153,3 +155,82 @@ def to_python(self, value):
+            raise ValidationError("Expected a list of URLs.")
+        return parsed
+    
+    def validate(self, value):
+        super().validate(value)
+        # Validate each URL
+        if value:
+            validator = URLValidator()
+            for url in value:
+                # Check type before calling .strip()
+                if not isinstance(url, str):
+                    raise ValidationError(f"Invalid URL type: expected string, got {type(url).__name__} ({url!r})")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)